backend/fms_core/services/index.py

```diff
@@ -0,0 +1,98 @@
+from fms_core.models import Index, IndexSet, IndexStructure, Sequence, SequenceByIndex3Prime, SequenceByIndex5Prime
+
+from django.core.exceptions import ValidationError
+
+
+def get_or_create_index_set(set_name):
+    index_set = None
+    created_entity = False
+    errors = []
+    warnings = []
+
+    if set_name:
+        try:
+            index_set = IndexSet.objects.get(name=set_name)
+
+        except IndexSet.DoesNotExist:
+            try:
+                index_set = IndexSet.objects.create(name=set_name)
+                created_entity = True
+            except ValidationError as e:
+                errors.append(';'.join(e.messages))
+
+    return index_set, created_entity, errors, warnings
+
+
+def create_index(index_name, index_structure, index_set=None):
+    index = None
+    errors = []
+    warnings = []
+
+    index_structure_obj = None
+    if index_structure:
+        try:
+            index_structure_obj = IndexStructure.objects.get(name=index_structure)
+        except IndexStructure.DoesNotExist:
+            errors.append(f"Invalid index structure.")
+            return index, errors, warnings
+
+        try:
+            index = Index.objects.create(index_set=index_set, name=index_name, index_structure=index_structure_obj)
+        except ValidationError as e:
+            errors.append(';'.join(e.messages))
+
+        return index, errors, warnings
+    else:
+        errors.append(f"Index structure is needed to create an index.")
+        return index, errors, warnings
```

Comment:
Mmm that fixes the issue but it would be cleaner to just take the return value out of the if else statement.